### DISCUSSION TITLE
http/1.1 "500 Internal Server Error ERROR:    Exception in ASGI application"

### DISCUSSION THREAD

**Discussion by JSPAnthony** [fastapi/fastapi#8373] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# import libraries
from fastapi import FastAPI
from pycaret.classification import *
import uvicorn
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

# define the app object
app = FastAPI()

# Load the model
model = load_model('rf_class_heart_failure_model')

# define function


@app.post('/predict')
def predict(age, sex, chest_pain, r_blood_pressure, cholesterol, f_blood_sugar, r_ecg, max_heart_rate, exer_angina,
            oldpeak, slope_peak, major_vessels, thal):
    data = pd.DataFrame([[age, sex, chest_pain, r_blood_pressure, cholesterol, f_blood_sugar, r_ecg, max_heart_rate,
                          exer_angina, oldpeak, slope_peak, major_vessels, thal]])
    data.columns = ['age', 'sex', 'chest_pain', 'r_blood_pressure', 'cholesterol', 'f_blood_sugar', 'r_ecg',
                    'max_heart_rate', 'exer_angina', 'oldpeak', 'slope_peak', 'major_vessels', 'thal']

    predictions = predict(model, data=data)
    print(data)
    return {'prediction': int(predictions['Label'][0])}


if __name__ == '__main__':
    uvicorn.run(app, host='127.0.0.1', port=8000)
```

### Description

I built a model which I want to deploy using fastapi, however when I run uvicorn main:app I get {"detail":"Not Found"} on 127.0.0.1:8000, and internal server error when I try to execute from 127.0.0.1:8000/docs.

<html>
<body>

Error: Internal Server ErrorResponse bodyDownloadInternal Server ErrorResponse headers content-length: 21  content-type: text/plain; charset=utf-8  date: Sat,20 Aug 2022 12:22:30 GMT  server: uvicorn
--

</body>
</html>

### Operating System

Windows

### Operating System Details

Am using Windows 11

### FastAPI Version

(env) PS D:\New\pycaret> pip show fastapi Name: fastapi Version: 0.79.1 Summary: FastAPI framework, high performance, easy to learn, fast to code, ready for production Home-page: https://github.com/tiangolo/fastapi Author: Sebastián Ramírez Author-email: tiangolo@gmail.com License: Location: d:\new\pycaret\env\lib\site-packages Requires: pydantic, starlette Required-by:

### Python Version

Python 3.8.8

**Comment by iudeen** [comment]
```text
{"detail":"Not Found"}
```
is expected when you hit on `127.0.0.1:8000`. It is because you do not have any route to `"/"`.

**Comment by iudeen** [comment]
As of `InternalServerError` it is most likely propagated from one of your internal methods. You can try encapsulating them into a `try` block to find what is the error. 

```python
try:
    predict(...)
except Exception as e:
    print(e)
```

**Comment by JSPAnthony** [comment]
@iudeen this is the error am now getting.

<html>
<body>


TypeError: Failed to execute 'fetch' on 'Window': Failed to parse URL from http://127.0.0.1:8000predict?age=70&sex=1&chest_pain=4&r_blood_pressure=130&cholesterol=322&f_blood_sugar=0&r_ecg=2&max_heart_rate=109&exer_angina=0&oldpeak=2.4&slope_peak=2&major_vessels=3&thal=3
--



</body>
</html>

**Comment by etimberg** [comment]
>  TypeError: Failed to execute 'fetch' on 'Window': Failed to parse URL from 

This error appears to be coming from JS. You should be calling `http://127.0.0.1:8000/predict?....`

**Comment by imacat** [comment]
Your function is named "predict":

    def predict(...

And you called "predict" in your "predict" function:

    predictions = predict(model, data=data)

It calls itself and causes infinite loop.  You should solve it and check again.

**Comment by jlramirez17** [comment]
conn.execute(oficina.insert().values(new_ofic))
AttributeError: 'APIRouter' object has no attribute 'insert'

At At record insert time it throws this error and doesn't insert anything into the table.
Your help please
